AppKit/CPWindow/CPWindow.j

```diff
@@ -1958,6 +1967,80 @@ CPTexturedBackgroundWindowMask
 
         case CPLeftMouseDragged:
         case CPRightMouseDragged:
+            // First, we search for any tracking area requesting CPTrackingEnabledDuringMouseDrag.
+            // At the same time, we update the entered stack.
+            var mouseEnteredStack = [];
+
+            for (var i = 0; i < _trackingAreaViews.length; i++)
+            {
+                var aView          = [_trackingAreaViews objectAtIndex:i],
+                    trackingAreas  = [aView trackingAreas],
+                    convertedPoint = [aView convertPoint:point fromView:_windowView];
```

Comment:
Since CPTrackingInVisibleRect is common usage, please cache the visible rect here to avoid calling `visibleRect` in the inner loop.